src/amb_kf.c

```diff
@@ -531,7 +531,7 @@ s32 find_index_of_element_in_u8s(const u32 num_elements, const u8 x, const u8 *l
       return i;
     }
   }
-  return -1;
+  assert(false);
```

Comment:
I'd leave this function returning `-1` for error and check the return code where its used. Also would be good to add a doxygen comment for the function and explaining the return value.